app/src/main/java/org/opencord/aaa/impl/RadiusOperationalStatusManager.java

```diff
@@ -0,0 +1,313 @@
+                } else if (statusServerReqSent && !serverStatusRadiusPacketDetails.
+                    isServerStatusPacketRecieved()) {
+                    setRadiusServerOperationalStatus(OperationalStatus.UNAVAILABLE);
+                }
+            } else {
+                if (radiusOperationalStatusEvaluationMode == RadiusOperationalStatusEvaluationMode.AUTO) {
+                    checkServerStatusUsingStatusServerRequest();
+                    if (statusServerReqSent && serverStatusRadiusPacketDetails.
+                        isServerStatusPacketRecieved()) {
+                        setRadiusServerOperationalStatus(OperationalStatus.IN_USE);
+                    } else {
+                        checkStatusServerForAccessRequestMode();
+                    }
+                } else {
+                    checkStatusServerForAccessRequestMode();
+                }
+            }
+            fakeAccessRequestRadiusPacketDetails.setFakeAccessRequestPacketRecieved(false);
+            serverStatusRadiusPacketDetails.setServerStatusPacketRecieved(false);
+        } catch (Exception e) {
+            log.error("Caught exception while checking radius server status::" + e);
+        }
+    }
+
+    /**
+     * Delegate allowing the RadiusOperationalStatus to notify us of events.
+     */
+     private class InternalRadiusOperationalStatusDelegate implements RadiusOperationalStatusEventDelegate {
+        @Override
+        public void notify(RadiusOperationalStatusEvent radiusOperationalStatusEvent) {
+            log.debug("Radius Operational Status event {} for {}", radiusOperationalStatusEvent.type(),
+                radiusOperationalStatusEvent.subject());
+            post(radiusOperationalStatusEvent);
+        }
+     }
+
+     class FakeAccessRequestRadiusPacketDetails {
+         boolean fakeAccessRequestPacketRecieved;
+         long fakeAccessRequestTimeInMillis;
+         long fakeAccessRequestOutTimeInMillis;
+
+         public long getFakeAccessRequestTimeInMillis() {
+             return fakeAccessRequestTimeInMillis;
+         }
+
+         public void setFakeAccessRequestTimeInMillis(long fakeAccessRequestTimeInMillis) {
+             this.fakeAccessRequestTimeInMillis = fakeAccessRequestTimeInMillis;
+         }
+
+         public long getFakeAccessRequestOutTimeInMillis() {
+             return fakeAccessRequestOutTimeInMillis;
+         }
+
+         public void setFakeAccessRequestOutTimeInMillis(long fakeAccessRequestOutTimeInMillis) {
+             this.fakeAccessRequestOutTimeInMillis = fakeAccessRequestOutTimeInMillis;
+         }
+
+         public boolean isfakeAccessRequestPacketRecieved() {
+             return fakeAccessRequestPacketRecieved;
+         }
+
+         public void setFakeAccessRequestPacketRecieved(boolean dummyPacketRecieved) {
+             this.fakeAccessRequestPacketRecieved = dummyPacketRecieved;
+         }
+     }
+
+     class ServerStatusRadiusPacketDetails {
```

Comment:
You are using this class only for keeping the 3 variables; better just define the variables outside in Manager class and remove this class. Same is for FakeAccessRequestRadiusPacketDetails  class